File: pdu/python/ackermann_msgs/pdu_cdr_conv_AckermannDriveStamped.py
from .pdu_pytype_AckermannDriveStamped import AckermannDriveStamped
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..ackermann_msgs.pdu_cdr_conv_AckermannDrive import *
from ..std_msgs.pdu_cdr_conv_Header import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_AckermannDriveStamped(writer: CdrWriter, src: AckermannDriveStamped):
    py_to_cdr_body_Header(writer, src.header)
    py_to_cdr_body_AckermannDrive(writer, src.drive)


def cdr_body_to_py_AckermannDriveStamped(reader: CdrReader, dst: AckermannDriveStamped):
    cdr_body_to_py_Header(reader, dst.header)
    cdr_body_to_py_AckermannDrive(reader, dst.drive)
    return dst


def py_to_cdr_AckermannDriveStamped(src: AckermannDriveStamped) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_AckermannDriveStamped(writer, src)
    return writer.bytes()


def cdr_to_py_AckermannDriveStamped(cdr_payload) -> AckermannDriveStamped:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = AckermannDriveStamped()
    return cdr_body_to_py_AckermannDriveStamped(reader, dst)

File: pdu/javascript/ackermann_msgs/pdu_cdr_conv_AckermannDriveStamped.js
import { AckermannDriveStamped } from './pdu_jstype_AckermannDriveStamped.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { AckermannDrive } from '../ackermann_msgs/pdu_jstype_AckermannDrive.js';
import { PduAckermannDriveConverter } from '../ackermann_msgs/pdu_cdr_conv_AckermannDrive.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduAckermannDriveStampedConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { AckermannDriveStamped } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        PduAckermannDriveConverter.to_cdr_body(writer, src.drive);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { AckermannDriveStamped } dst
     * @returns { AckermannDriveStamped }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduAckermannDriveConverter.cdr_body_to_js(reader, dst.drive);
        return dst;
    }

    /**
     * @param { AckermannDriveStamped } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { AckermannDriveStamped }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new AckermannDriveStamped());
    }
}
